go/apps/gw/router/gateway_proxy/handler.go

```diff
@@ -0,0 +1,82 @@
+package gateway_proxy
+
+import (
+	"context"
+
+	"github.com/unkeyed/unkey/go/apps/gw/server"
+	"github.com/unkeyed/unkey/go/apps/gw/services/auth"
+	"github.com/unkeyed/unkey/go/apps/gw/services/proxy"
+	"github.com/unkeyed/unkey/go/apps/gw/services/routing"
+	"github.com/unkeyed/unkey/go/apps/gw/services/validation"
+	"github.com/unkeyed/unkey/go/pkg/codes"
+	"github.com/unkeyed/unkey/go/pkg/fault"
+	"github.com/unkeyed/unkey/go/pkg/otel/logging"
+	"github.com/unkeyed/unkey/go/pkg/otel/tracing"
+)
+
+// Handler implements the main gateway proxy functionality.
+// It handles all incoming requests and forwards them to backend services.
+type Handler struct {
+	Logger         logging.Logger
+	RoutingService routing.Service
+	Proxy          proxy.Proxy
+	Auth           auth.Authenticator
+	Validator      validation.Validator
+}
+
+// Handle processes all HTTPS requests for the gateway.
+// This is the main entry point for the gateway - all requests go through here.
+func (h *Handler) Handle(ctx context.Context, sess *server.Session) error {
+	ctx, span := tracing.Start(ctx, "proxy.handler")
+	defer span.End()
+	req := sess.Request()
+
+	// Look up target configuration based on the request host
+	// Strip port from hostname for database lookup (Host header may include port)
+	hostname := routing.ExtractHostname(req)
+
+	config, err := h.RoutingService.GetConfig(ctx, hostname)
+	if err != nil {
+		return fault.Wrap(err,
+			fault.Code(codes.Gateway.Routing.ConfigNotFound.URN()),
```

Comment:
@coderabbitai please create an issue for us to doublecheck that we have documented all of these new gateway codes in our public documentation